### PR TITLE
Build support library from base and context.

### PR DESCRIPTION
**Pull request by mpreiner**
This PR creates a support library from the utilities in base and context, which will be required in the parser as soon as we move the symbol table/manager to the parser.

Note: I decided to always build static libraries from base and context (and optionally enable -fPIC for shared builds) since I'm not sure if we want to have these libraries installed separately. Right now these are considered as cvc5 internal utilities that can be used in all cvc5 libraries, but not outside.

### CODE REVIEW

File: src/base/CMakeLists.txt [hunk 1]
```diff
@@ -75,3 +78,14 @@ set_source_files_properties(
...
+
+add_library(cvc4base STATIC ${LIBBASE_SOURCES})
+if(ENABLE_SHARED)
+  set_target_properties(cvc4base PROPERTIES POSITION_INDEPENDENT_CODE ON)
+endif()
+target_compile_definitions(cvc4base
+  PRIVATE
+    -D__BUILDING_CVC4LIB
+    -D__STDC_LIMIT_MACROS
+    -D__STDC_FORMAT_MACROS
```
Comment (nafur) [inline src/base/CMakeLists.txt:90]: What do these do?

Comment (mpreiner) [reply src/base/CMakeLists.txt:90]: That's what I've been wondering too. It is an old relic from the old build infrastructure, maybe we can get rid of it.

Comment (mpreiner) [reply src/base/CMakeLists.txt]: I removed it.

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**